src/main/java/com/google/cloud/spanner/pgadapter/Server.java

```diff
@@ -31,13 +45,87 @@ public class Server {
   public static void main(String[] args) {
     try {
       OptionsMetadata optionsMetadata = extractMetadata(args, System.out);
-      ProxyServer server = new ProxyServer(optionsMetadata);
+      OpenTelemetry openTelemetry = setupOpenTelemetry(optionsMetadata);
+      ProxyServer server = new ProxyServer(optionsMetadata, openTelemetry);
       server.startServer();
     } catch (Exception e) {
       printError(e, System.err, System.out);
     }
   }
 
+  static OpenTelemetry setupOpenTelemetry(OptionsMetadata optionsMetadata) throws IOException {
+    if (!optionsMetadata.isEnableOpenTelemetry()) {
+      return OpenTelemetry.noop();
```

Comment:
Why is this noop?

Reply:
Returning a `noop` OpenTelemetry instance instead of null when OpenTelemetry has not been enabled makes the rest of the code simpler, as we don't need null checks everywhere.

Reply:
Thanks!